backend/src/tasks/LookingGlass.ts

```diff
@@ -0,0 +1,286 @@
+    const organization = await Organization.findOne(organizationId);
+    if (organization) {
+      return organization;
+    } else {
+      console.log('No organization found');
+      exit;
+    }
+  } else {
+    console.log('No OrganizationId provided');
+    exit;
+  }
+}
+
+async function collectionByWorkspace() {
+  const resource =
+    'https://delta.lookingglasscyber.com/api/v1/workspaces/' +
+    'NCATS POV' +
+    '/collections';
+
+  return (await got(resource, {
+    headers: { Authorization: 'Bearer ' + process.env.LG_API_KEY }
+  }).json()) as LGCollectionsResponse[];
+}
+
+function ValidateIPaddress(ipaddress) {
+  if (/\b\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\b/.test(ipaddress)) {
+    return true;
+  }
+  return false;
+}
+
+async function getThreatInfo(collectionID) {
+  const resource = 'https://delta.lookingglasscyber.com/api/graph/query';
+  const modifier = {
+    query: [
+      'and',
+      ['=', 'type', ['associated-with']],
+      ['or', ['=', 'right.type', 'threat'], ['=', 'left.type', 'ipv4']],
+      ['=', 'left.collectionIds', collectionID]
+    ],
+    fields: ['firstSeen', 'lastSeen', 'sources', 'right', 'left'],
+    limit: 100000,
+    workspaceIds: []
+  };
+  const h1 = {
+    'Content-Type': 'application/json',
+    Authorization: 'Bearer ' + process.env.LG_API_KEY
+  };
+  return (got
+    .post(resource, {
+      json: modifier,
+      headers: h1
+    })
+    .json() as unknown) as LGThreatResponse[];
+}
+
+async function saveAndPullDomains(response, organizationId, scanId, Org) {
+  const domains: Domain[] = [];
+  const data = response;
+  for (const l of data['results']) {
+    if (ValidateIPaddress(l['left']['name'])) {
+      const current_Domain = plainToClass(Domain, {
+        name: l['left']['name'],
+        ip: l['left']['name'],
+        organization: { id: organizationId },
+        fromRootDomain: Org['rootDomains'][0],
+        ipOnly: true,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(current_Domain);
+    } else {
+      const current_Domain = plainToClass(Domain, {
+        name: l['left']['name'],
+        ip: l['left']['name'],
+        organization: { id: organizationId },
+        fromRootDomain: Org['rootDomains'][0],
+        ipOnly: false,
+        discoveredBy: { id: scanId }
+      });
+      domains.push(current_Domain);
+    }
+  }
+  await saveDomainsToDb(domains);
+
+  await connectToDatabase();
+
+  let pulledDomains = Domain.createQueryBuilder('domain')
+    .leftJoinAndSelect('domain.organization', 'organization')
+    .andWhere('ip IS NOT NULL');
+
+  if (organizationId) {
+    pulledDomains = pulledDomains.andWhere('domain.organization=:org', {
+      org: organizationId
+    });
+  }
+  const D = await pulledDomains.getMany();
+  return D;
+}
+
+export const handler = async (commandOptions: CommandOptions) => {
+  const { organizationId, organizationName, scanId } = commandOptions;
+
+  const Org = await getOrg(organizationId);
```

Comment:
We should remove this line and remove the getOrg function. We don't actually use the Org variable anywhere else.
```suggestion
```

Reply:
Right now it is using the Org to assign the FromRootDomain when creating the domains based off of ip addresses since we can't really verify which root domain it is associated with. Is the fromRootDomain value required?, if not we can leave it blank and get rid of the Org

Reply:
Oh, you're right, thanks for catching that. Let's rename `Org` to `organization`, though.

```suggestion
  const organization = await getOrg(organizationId);
```

Reply:
Actually, my bad. We should remove this line, because you're right, we should set fromRootDomain to be blank instead. fromRootDomain was originally created in order to store which domain from findomain created the domain; but in this case, the domain isn't even created from findomain.